homeassistant/components/media_player/epson.py

```diff
@@ -0,0 +1,316 @@
+"""
+Support for Epson projector.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/media_player.epson/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+
+import aiohttp
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    DOMAIN, MEDIA_PLAYER_SCHEMA, PLATFORM_SCHEMA, SUPPORT_NEXT_TRACK,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_SELECT_SOURCE, SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_STEP,
+    MediaPlayerDevice)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, CONF_HOST, CONF_NAME, CONF_PORT, CONF_SSL, STATE_OFF,
+    STATE_ON, STATE_UNKNOWN)
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+import homeassistant.helpers.config_validation as cv
+
+KEY_COMMANDS = {
+    "TURN_ON": [('KEY', '3B')],
+    "TURN_OFF": [('KEY', '3B'), ('KEY', '3B')],
+    "HDMILINK": [('jsoncallback', 'HDMILINK?')],
+    "CMODE": [('jsoncallback', 'CMODE?')],
+    "CMODE_CINEMA": [('CMODE', '15')],
+    "CMODE_NATURAL": [('CMODE', '07')],
+    "CMODE_BRIGHT": [('CMODE', '0C')],
+    "CMODE_DYNAMIC": [('CMODE', '06S')],
+    "VOL_UP": [('KEY', '56')],
+    "VOL_DOWN": [('KEY', '57')],
+    "MUTE": [('KEY', '3E')],
+    "HDMI1": [('KEY', '4D')],
+    "HDMI2": [('KEY', '40')],
+    "PC": [('KEY', '44')],
+    "VIDEO": [('KEY', '46')],
+    "USB": [('KEY', '85')],
+    "LAN": [('KEY', '53')],
+    "WFD": [('KEY', '56')],
+    "PLAY": [('KEY', 'D1')],
+    "PAUSE": [('KEY', 'D3')],
+    "STOP": [('KEY', 'D2')],
+    "BACK": [('KEY', 'D4')],
+    "FAST": [('KEY', 'D5')],
+}
+
+DEFAULT_SOURCES = {
+    'HDMI1': 'HDMI1',
+    'HDMI2': 'HDMI2',
+    'PC': 'PC',
+    'VIDEO': 'VIDEO',
+    'USB': 'USB',
+    'LAN': 'LAN',
+    'WFD': 'WiFi Direct'
+}
+
+INV_SOURCES = {v: k for k, v in DEFAULT_SOURCES.items()}
+
+CMODE_LIST = {
+    '15': 'Cinema',
+    '07': 'Natural',
+    '0C': 'Bright Cinema',
+    '06': 'Dynamic'
+}
+
+CMODE_LIST_SET = {
+    'cinema': 'CMODE_CINEMA',
+    'natural': 'CMODE_NATURAL',
+    'bright cinema': 'CMODE_BRIGHT',
+    'dynamic': 'CMODE_DYNAMIC'
+}
+
+DATA_EPSON = 'epson'
+DEFAULT_NAME = 'EPSON Projector'
+
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=80): cv.port,
+    vol.Optional(CONF_SSL, default=False): cv.boolean
+})
+ATTR_CMODE = 'cmode'
+SUPPORT_CMODE = 33001
+ACCEPT_ENCODING = "gzip, deflate"
+ACCEPT_HEADER = "application/json, text/javascript"
```

Comment:
Please use the constants from `from aiohttp import hdrs` or `const.py` if possible.